src/rtl/hbsocket.c

```diff
@@ -2848,6 +2860,21 @@ int hb_socketSetNoDelay( HB_SOCKET sd, HB_BOOL fNoDelay )
    return ret;
 }
 
+int hb_socketSetNoSigPipe( HB_SOCKET sd, HB_BOOL fNoSigPipe )
+{
+#if defined( SO_NOSIGPIPE )
```

Comment:
isn't this condition always true due to #ifndef...#define above?

Reply:
No, if not defined HB_SOCKET_OFF then the SO_NOSIGPIPE is not defined too.